wasm-rpc-stubgen/src/wit.rs

```diff
@@ -171,6 +433,28 @@ pub fn copy_wit_files(def: &StubDefinition) -> anyhow::Result<()> {
 
             fs::create_dir_all(&dep_dir)?;
             for source in unresolved.source_files() {
+                // While copying this root WIT (from which the stub is generated) to the deps directory
+                // of the stub, we ensure there is no import of the same stub's types. This happens
+                // when you regenerate a stub from a WIT, which the user may have manually modified to refer to the same stub.
+                // This occurs only when there is a cyclic dependency. That's when the package from which the stub is generated is using the stub itself.
+                //
+                // Replacing this import with `none` is closer to being the right thing to do at this stage,
+                // because there is no reason a stub WIT needs to cyclically depend on its own types to generate the stub.
+                // Now the question is whether we need to parse this WIT to Resolved and then have our own Display
+                // for it, or just use regex replace. The latter is chosen because we hardly refer to the same stub without
+                // `import`, and the actual stub package name.
+                // The former may look cleaner, but slower, and may not bring about anymore significant correctness or reliability.
```

Comment:
I don't think being slower is a good argument, there is no need to optimize the stub generator for performance

Reply:
Yea, not biggest of the reason